### DISCUSSION TITLE
Matek F722mini gps and esc telem issue

### DISCUSSION THREAD

**Discussion by lethalcobra** [iNavFlight/inav#6633] [Q&A]
Hi,

As I reported my problems in detail [here](https://www.rcgroups.com/forums/showthread.php?3826089-INAV-2-6-MATEK-GPS-and-ESC-telemetry-problem),  I am building a 4" LR style quad with Matek f722mini FCU with INAV 2.6.
## Current Behavior

 I have connected a Matek GPS-compass unit to it, the serial and I2C connection is working, but the compass is heavily drifting in random directions, the GPS is very uncertain, can rarely get any fiy (2D and 3D either). Also, I am trying to use ESC telemetry on UART RX6, as supposed, but no telemetry is received. 

Downgrading to 2.5.X makes the GPS working, but this FW does not support the barometer on Matek F722mini. ESC telemetry is still not working.

## Steps to Reproduce

1. Use INAV 2.6.0
2. GPS on UART 3
3. ESC telem: UART 6 RX

## Expected behavior
GPS fix in minutes after switch on, ESC data to receive.

## Suggested solution(s)
There was something different in the GPS handling in INAV 2.5.X that made my GPS work perfectly but changed in INAV 2.6.0... Is it possible to use the older handling method, please?

- FC Board name and vendor: Matek F722mini
- INAV version string: 2.6.0

Thank you, all the best, and keep flying!

Csaba Kovago, Hungary

**Comment by DzikuVx** [comment]
Because I have this FC, this GPS and INAv 2.6 I can confirm that it's working exactly like it should be. 
Please remember that you have to calibrate the magnetometer after upgrade from 2.5 to 2.6

**Comment by lethalcobra** [comment]
Dear Pawel,

Thank you for your email!

First, I am happy to meet with you, I have seen your videos about INAV,
thank you for them as well!

So, if you happen to read my complete description about my problem on
RCGroups forum (yes, it is rather long, but I wanted to give all the
details), please help me, what do I do wrong?

I installed the 2.6 as it should, I did the "default" in CLI, and I set up
all the ports, (GPS, RX, opt. flow, ESC telem) baud-s are default (115200
if I recall it correctly).

I had some problems with the GPS, I corrected it in the uBlox Central.
Still, if I connect the battery, it needs about 10 minutes to find some
sats... If I connect the GPS to the PC via FTDI, it finds fix (on the same
spot in the room) in 1-2 mins...

Magnetometer was calibrated, opt flow was calibrated, etc. I set up the ESC
telem. as you showed in  your video (RX6, protocol DShot 600), but no data
are coming in...

I can send you the "dump" if it is needed!

Thank you in advance,

Csaba Kovago
